src/main/java/dk/cngroup/trainings/spring/springassignment/controller/advice/RESTControllerException.java

```diff
@@ -3,11 +3,11 @@
 import dk.cngroup.trainings.spring.springassignment.exception.*;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
-import org.springframework.web.bind.annotation.RestController;
 
-@ControllerAdvice(annotations = RestController.class)
+@ControllerAdvice(annotations = Controller.class)
```

Comment:
Ok, now you have some kind of error handling, but would not it be better if you actually got the error message where it happened? E.g. when you add new animal and the name is too short, it should show the form again (with the values you sent) and with some error message above or below the form. Don't you think?

Reply:
Yeah, I've that in mind, I just wanted to implement the basic functionalities then I will head for error handling. An Yes you're right. error of fields validation should return to the same form.

Reply:
Yeah, I've that in mind. I wanted to finish all basic functionalities first then head to error handling.
You're right, error in fields validation should return to the same form.